src/test/java/com/savi/geo/timezones/TzWhereTest.java

```diff
@@ -0,0 +1,126 @@
+package com.savi.geo.timezones;
+
+import org.junit.*;
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.*;
+
+
+public class TzWhereTest {
+
+    @Test
+    public void testSomePointGetsItsTimeZoneId() {
+        final double latitude  = 38.8044;
+        final double longitude = -77.0920;
+        final String actual = TzWhere.getLatLongTimeZoneId(latitude, longitude);
+        final String expected = "America/New_York";
+
+        assertThat(actual, equalTo(expected));
+    }
+
+    @Test
+    public void testPointInCollidedTileGetsRightTimeZone() {
+        // A point in an index tile(?) that maps to two timezone portions and
+        // needs resolution via the actual geometry (as of 2016-05):
+        // (37.952126 N / -86.764727) is in America/Indiana/Tell_City but
+        // America/Chicago also gets indexed into same internal tile.
+        final double latitude  = 37.952126;
+        final double longitude = -86.764727;
+        final String actual = TzWhere.getLatLongTimeZoneId(latitude, longitude);
+        final String expected = "America/Indiana/Tell_City";
+    }
+
+    @Test
+    public void testPointAtBoundaryVertex() {
+        // A point that is vertex on boundary polygon of Europe/Vatican:
+        // (Attempted exact values--but unconfirmed.)
+        final double latitude  = 41.9063606262207;
+        final double longitude = 12.45091629028328;
```

Comment:
Wow, that's super precise. As to your statement in `getLatLongTimeZoneId`, I think we just need to be clear about what we expect to be able to provide. I think if we somehow hit a point that is legitimately inside two geometries, returning the last one is a sensible as anything else. It's going to be very hard to test all of those cases, and it will likely be easier to let reported issues dictate how we handle it in a more graceful way.